src/VisualStudio/Core/Def/Implementation/Library/ObjectBrowser/AbstractObjectBrowserLibraryManager.cs

```diff
@@ -76,7 +76,9 @@ public void Dispose()
                     var oldDocument = e.OldSolution.GetDocument(e.DocumentId);
                     var newDocument = e.NewSolution.GetDocument(e.DocumentId);
 
-                    await DocumentChangedAsync(oldDocument, newDocument).ConfigureAwait(false);
+                    // make sure we do this in background thread. we don't care about ordering of events
+                    // we just need to refresh OB at some point if it ever needs to be updated
+                    Task.Run(() => DocumentChangedAsync(oldDocument, newDocument));
```

Comment:
@Pilchie oh, didn't know Task.Run will swallow exception. didn't task with unobserved exception crash process in finalizer?

I can add comment for the bug, but I couldn't find the bug from github that is assigned to  @jasonmalinowski , can you give me url for the bug?

...

regardless of root cause, async void looks dangerous. Task.Run or await only difference should be whether there is a chance whether task can be inlined (run in same thread as await is called) or always run from BG.

if exception is the issue, we can add await Task.Run(() ....).ConfigureAwait(false); I believe so that exception can propagate.